pandas/core/categorical.py

```diff
@@ -308,6 +308,9 @@ def __init__(self, values, categories=None, ordered=None, dtype=None,
         elif isinstance(values, (ABCIndexClass, ABCSeries)):
             # we'll do inference later
             pass
+        elif isinstance(values, ABCDataFrame):
```

Comment:
instead of this specific tests, rather check the ndim of the passed values, it must be 1 (or None)

e.g.
```getattr(values, 'ndim', None) in [1, None]```